### ISSUE TITLE
autocomplete not working with angular-cli project

### ISSUE THREAD

**Issue by mickeychu** [TypeStrong/atom-typescript#1043]
I'm using angular-cli for my current project. It works fine with JS and other files but when I move to ts file, the autocomplete is totally not working. 

I tried to delete the tsconfig.json file and regenerate with _Typescript: create Tsconfig.json project file_ but no file created.

Tested with another project without angular-cli, autocomplete works fine.

Any suggestion to fix this probem?

Many thanks!!!

**Comment by abizit** [comment]
@mickeychu did you find any solution?? I am having the same problem. Using Angular cli and the autocomplete is very very slow.

**Comment by mickeychu** [comment]
@abizit : after hours searching,  I end up with this tsconfig.json. Now the autocomplete works ok for me. Hope it can help you or anybody with the same problem.

`{
  "compileOnSave": false,
  "compilerOptions": {
    "declaration": false,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "module": "commonjs",
    "moduleResolution": "node",
    "noEmitOnError": true,
    "noImplicitAny": false,
    "outDir": "../dist/",
    "rootDir": ".",
    "sourceMap": true,
    "target": "es5",
    "inlineSources": true
  },
  "filesGlob": [
    "**/*.ts"
  ],
  "exclude": [
    "./dist/**/*",
    "./node_modules/**/*",
    "./typings/browser",
    "./typings/browser.d.ts",
    "*.spec.ts"
  ],
  "atom": {
    "rewriteTsconfig": true
  },
  "files": [
    "app/+dashboard/dashboard.component.spec.ts",
    "app/+dashboard/dashboard.component.ts",
    "main.ts",
    "system-config.ts",
    "typings.d.ts"
  ]
}
`

**Comment by arlowhite** [comment]
Is your tsconfig.json file at **src/tsconfig.json**?

I was under the impression you didn't need worry about excluding directories above where tsconfig.json is.

**Comment by anderflash** [comment]
Same problem, still very slow and it's tsconfig is not correct for recent Angular-CLI projects (`node_modules` is one level above `src/`, not current one). I tried to "exclude" `../node_modules`, but it doesn't work.

**Comment by cuongnm265** [comment]
same issue here. It make me very confused

**Comment by jermainerf** [comment]
Same issue, very slow. After every letter I type it's 5-10 seconds before autocomplete pops up. I tried the official angular quickstart (without CLI) from angular.io and I have no wait at all. So I compared the tsconfig settings one by one and changed them to match and I found that removing:
    "lib": ["es6", "dom"],
fixed this right up. I don't pretend to know what removing that line really does in the grand scheme of things but I was able to use ng serve no problem.
Hopefully that helps someone else solve this problem.
